src/redux/reducers/samples/samplesFileUpdate.js

```diff
@@ -1,14 +1,36 @@
+import _ from 'lodash';
+
 const samplesFileUpdate = (state, action) => {
-  const { sampleUuid, file, lastModified } = action.payload;
+  const {
+    sampleUuid, fileName, fileDiff, lastModified,
+  } = action.payload;
+
+  const overwriteIfArray = (objValue, srcValue) => {
+    if (_.isArray(objValue) && srcValue) {
+      return srcValue;
+    }
+  };
+
+  const oldFile = state[sampleUuid].files?.[fileName];
+  let newFile = fileDiff;
+
+  if (oldFile) {
+    newFile = _.mergeWith(oldFile, fileDiff, overwriteIfArray);
+  }
+
+  newFile.lastModified = lastModified;
+
+  const newFileNames = _.clone(state[sampleUuid].fileNames);
+  newFileNames.add(fileName);
 
   return {
     ...state,
     [sampleUuid]: {
       ...state[sampleUuid],
-      fileNames: [...state[sampleUuid].fileNames, file.name],
+      fileNames: newFileNames,
```

Comment:
A more concise way to write this can be `new Set([...state[sampleUuid].fileNames, fileName])`